internal/server/server.go

```diff
@@ -97,7 +97,7 @@ func (ts *TerminalServer) Run(ctx context.Context) (err error) {
 			Handler: http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 				contentType := r.Header.Get("content-type")
 				switch {
-				case grpcWebServer.IsGrpcWebSocketRequest(r):
+				case strings.ToLower(r.Header.Get("Sec-Websocket-Protocol")) == "grpc-websockets":
```

Comment:
`IsGrpcWebSocketRequest` is doing pretty much the same except it also checks for `Upgrade` header. Are you sure this is the proper way?

https://github.com/improbable-eng/grpc-web/blob/84ab65f9526bd73430fb786dced98135186dd099/go/grpcweb/wrapper.go#L136

Reply:
> Are you sure this is the proper way?

When the client is using HTTP/2 the `Connection` header gets stripped for some reason and it's unlikely that `Sec-Websocket-Protocol` is used in other cases we check for, so yes.